Sprint 112 blog post: packages and requests into the watchlist (#344)
openSUSE/obs-landing: _posts/development/2022-02-22-notifications-watchlist-packages-requests.md

---
layout: post
title: Packages and Requests arrive to your Watchlist
category: development
---

> Your watchlist, where you kept your most used projects until now, now supports packages and requests.

Reviewer: ```suggestion
Your watchlist, where you keep your most used projects, now supports packages and requests.
```

Reviewer: Just a suggestion, but twice `now` sound a bit weird in my opinion. Also you still keep your projects in the watchlist :)

OBS is huge.
And why not say it straight: it is easy to lose track of what matters.

> The watchlist came to solve this making it possible to add your favorite projects to a list, available from every corner of the application.

Reviewer: ```suggestion
The watchlist came to solve this by making it possible to add your favorite projects to a list, available from every corner of the application.
```

But it was only possible to add projects.

> We are happy to introduce managing packages and requests to your watchlist.

Reviewer: ```suggestion
We are happy to introduce managing packages and requests through your watchlist.
```

An image will make it clear.

<figure>
  <img src="/images/posts/sprint_112/watchlist_with_packages_and_requests.png" alt="Watchlist with packages and requests" />
  <figcaption>Watchlist with packages and requests</figcaption>
</figure>

> Whenever you want to add a package your watchlist, do the same as with projects: browse to that package, click on the watchlist icon, and add that package.

Reviewer: ```suggestion
Whenever you want to add a package to your watchlist, do the same as with projects: browse to that package, click on the watchlist icon, and add that package.
```

Reviewer: last thing I found, besides that LGTM :)

Author: Ops, fixed!

The same applies to a request.

> Awesome! Isn't it? ;-)

Reviewer 2: Do you really need to make fun of your own work? :)

Author: That was not the intention. I only wanted to make the blog post more catchy, not so boring. What are you suggesting to change?

Reviewer 2: Remove it? Maybe move the `This improvement will open the door` sentence up here...

Author: Understood. I removed that line and moved the. `This improvement will open the door...` sentence up.

# Stay Tuned

This improvement will open the door in the future to receive notifications about changes related to what you keep in your watchlist.

> This new capability is only reachable under [the beta program](/2018/10/04/the-beta-program/).

Author: Suggestion applied.

We kindly ask you to join the beta program, test it, and provide feedback.

> Just be aware that you will keep two separate watchlists: one legacy and one for the beta.

Reviewer: ```suggestion
Just be aware that for now you will keep two separate watchlists: one legacy and one for the beta program.
```

Reviewer 3: @krauselukas I agree with your suggestion. I edited your suggestion to add a missing comma.

When this feature rolls out from beta, you will not lose anything.

> Your definitive watchlist will contain the projects you watched in the legacy watchlist.

Reviewer 3: In the previous sentences, we mention that there is a legacy watchlist and one in the beta program. The definitive is the merge of both. Is this not clear enough?

> Your definitive watchlist will contain the projects you watched in the legacy watchlist.

Reviewer 2: You mean like the one after merging? You already promised this will not change, I think you can just drop this sentence.

> Your definitive watchlist will contain the projects you watched in the legacy watchlist.

Reviewer 2: Okay maybe this is just me: non-native-speaker

> Your definitive watchlist will contain the projects you watched in the legacy watchlist.

Reviewer 4: I don't know if I understand... what did we promise?

Regarding the data, we decided not to do an initial data migration (not now). We said it is not a problem if the users in the beta see an empty list.

But we planned to do a migration as soon as we roll the feature out.

> Your definitive watchlist will contain the projects you watched in the legacy watchlist.

Reviewer 2: Your promise:

> When this feature rolls out from beta, you will not lose anything.

> Your definitive watchlist will contain the projects you watched in the legacy watchlist.

Reviewer 4: Yes, that's the idea. We'll move all the projects the user had in the legacy watchlist to the new one (avoiding duplications and not removing anything).

How can we _not_ fulfil what we promised?
